src/mlpack/methods/det/dt_utils_impl.hpp

```diff
@@ -18,23 +18,23 @@
 namespace mlpack {
 namespace det {
 
-template <typename MatType, typename TagType>
-void PrintLeafMembership(DTree<MatType, TagType>* dtree,
+template <typename MatType>
+void PrintLeafMembership(DTree<MatType, int>* dtree,
```

Comment:
Because the the output of `FindBucket()` is used for index in a table:
```
...
    const int leafTag = dtree->FindBucket(testPoint);
    const size_t label = labels[i];
    table(leafTag, label) += 1;
...
```

It's not fair to pretend we're supporting other type of tagging for this functionality.

Reply:
Ah, ok.  In this case should we remove the `TagType` template argument entirely then?  Alternately as a workaround, we can have a static_assert requiring that `TagType` is `int` in the `DTree` class definition, then open an issue to de-templatize the tag.  (If I remember right, the `TagType` support itself came from you, so if you want to see it go a different direction that's ok too.  It would probably be useful to open an issue for it either way if you don't want to address it here.)